docs/recipes/install/rocky9/x86_64/warewulf4/slurm/steps.tex

```diff
@@ -0,0 +1,298 @@
+
+\subsection{Enable \OHPC{} repository for local use} \label{sec:enable_repo}
+\input{common/enable_ohpc_repo}
+\input{common/rocky_repos}
+\input{common/automation}
+
+
+\subsection{Add provisioning services on {\em master} node} \label{sec:add_provisioning}
+\input{common/install_provisioning_warewulf4_intro}
+\input{common/enable_pxe}
+\input{common/time}
+
+\vspace*{0.15cm}
+\subsection{Add resource management services on {\em master} node} \label{sec:add_rm}
+\input{common/install_slurm}
+
+\subsection{Optionally add \InfiniBand{} support services on {\em master} node} \label{sec:add_ofed}
+\input{common/ibsupport_sms_centos}
+
+\subsection{Optionally add \OmniPath{} support services on {\em master} node} \label{sec:add_opa}
+\input{common/opasupport_sms_centos}
+
+\vspace*{-0.15cm}
+\subsection{Complete basic Warewulf setup for {\em master} node} \label{sec:setup_ww}
+\input{common/warewulf4_setup}
+\input{common/warewulf4_setup_centos}
+
+\subsection{Define {\em compute} image for provisioning}
+\input{common/warewulf4_mkchroot_rocky}
+
+\subsubsection{Add \OHPC{} components} \label{sec:add_components}
+\input{common/warewulf4_add_to_compute_chroot_intro}
+
+%\newpage
+% begin_ohpc_run
+% ohpc_validation_comment Add SLURM and other components to compute instance
+\begin{lstlisting}[language=bash,literate={-}{-}1,keywords={},upquote=true,literate={BOSVER}{\baseos{}}1]
+[sms](*\#*) wwctl container exec rocky-9.4 /bin/bash <<- EOF 
+  # Add Slurm client support meta-package and enable munge and slurmd
+  dnf -y install ohpc-slurm-client
+  systemctl enable munge
+  systemctl enable slurmd
+
+  # Add Network Time Protocol (NTP) support
+  dnf -y intall chrony
+
+  # Include modules user environment
+  dnf -y intall lmod-ohpc
+
+  # defer image rebuild
+  /bin/false
+EOF
+\end{lstlisting}
+% end_ohpc_run
+
+\vspace*{.2cm}
+\subsubsection{Customize system configuration} \label{sec:master_customization}
+\input{common/oneapi_mountpoint}
+
+% Additional commands when additional computes are requested
+
+% begin_ohpc_run
+% ohpc_validation_newline
+% ohpc_validation_comment Update basic slurm configuration if additional computes defined
+% ohpc_command if [ ${num_computes} -gt 4 ];then
+% ohpc_command    perl -pi -e "s/^NodeName=(\S+)/NodeName=${compute_prefix}[1-${num_computes}]/" /etc/slurm/slurm.conf
+% ohpc_command    perl -pi -e "s/^PartitionName=normal Nodes=(\S+)/PartitionName=normal Nodes=${compute_prefix}[1-${num_computes}]/" /etc/slurm/slurm.conf
+
+% ohpc_command fi
+% end_ohpc_run
+
+%\clearpage
+\subsubsection{Additional Customization ({\em optional})} \label{sec:addl_customizations}
+\input{common/compute_customizations_intro}
+
+%\clearpage
+\paragraph{Enable \InfiniBand{} drivers}
+\input{common/ibsupport_compute_centos.tex}
+
+\paragraph{Enable \OmniPath{} drivers}
+\input{common/opasupport_compute_centos.tex}
+
+\vspace*{0.28cm}
+\paragraph{Increase locked memory limits}
+\input{common/memlimits}
+
```

Comment:
The warewulf 3 recipe includes the following here:
```
\vspace*{-.17cm}
\paragraph{Enable ssh control via resource manager}
\input{common/slurm_pam}
```

Reply:
I dropped that as it's not necessary, Warewulf 4 manages ssh keys.

Reply:
I am not convinced. The reason for the slurm PAM module is to allow SSH login to nodes which are assigned to the user via slurm. Users should not be able to login to nodes which are not part of the selection done by slurm.

Reply:
This is also still needed. I can login into nodes currently even if they are not assigned by slurm to my user.

Reply:
Added `slurm_pam` back - not sure why you need it but it "works for me" without it.

Reply:
Can you define works for me?

Reply:
```
[root@head ~]# cat .ssh/authorized_keys 
ssh-rsa AAAAB3NzaC1yc2EAAAADAQABAAABgQCKlnQlK2IIzKtCO39YEPfxSllpF/N32spe6Cepz8khCKbcci7oMsb0CgG8MQR3PwJyWza3kvxtwLhQOBwN2MtFDHnOQH3Tq0ekbvekLmYUA+SvLiCiZGObkHqYhjAfZ7DkIxBSyQvFASF8KT8j75XFQHyXvhwkc7x45Duia4sl1GOViVLdQ+IvkOmD2MclNHQgod9ofYd94fs8AbL6iVSDCxuohHQ9NtXd844CyI43RW2cIeXOINwCVXF25Ir+TR4OcC0UcmGnBEFIWXPu91znteW1MmZfFoy8brpJAQgySkseqFjdd0h16rXvRghQxn3ccSV1nbpjdt9h9l+kdBnFxUkYKN6jTOrVPJz6gDORgYA4pzXcIJuzeK4snh6EGKWhYP+r428qxiAp8sR18a8gSS6w9v97HuUOUjoNw/7UAXmVIOO9W7+EqEfdCCiBFye2dKY0GG31RcDDinXeUa3aT/k5cqmsMv6BLR83L0JWTKJd4Z4NTtyjQnfTCgSjTTM= 
[root@head ~]# cat .ssh/id_rsa.pub 
ssh-rsa AAAAB3NzaC1yc2EAAAADAQABAAABgQCKlnQlK2IIzKtCO39YEPfxSllpF/N32spe6Cepz8khCKbcci7oMsb0CgG8MQR3PwJyWza3kvxtwLhQOBwN2MtFDHnOQH3Tq0ekbvekLmYUA+SvLiCiZGObkHqYhjAfZ7DkIxBSyQvFASF8KT8j75XFQHyXvhwkc7x45Duia4sl1GOViVLdQ+IvkOmD2MclNHQgod9ofYd94fs8AbL6iVSDCxuohHQ9NtXd844CyI43RW2cIeXOINwCVXF25Ir+TR4OcC0UcmGnBEFIWXPu91znteW1MmZfFoy8brpJAQgySkseqFjdd0h16rXvRghQxn3ccSV1nbpjdt9h9l+kdBnFxUkYKN6jTOrVPJz6gDORgYA4pzXcIJuzeK4snh6EGKWhYP+r428qxiAp8sR18a8gSS6w9v97HuUOUjoNw/7UAXmVIOO9W7+EqEfdCCiBFye2dKY0GG31RcDDinXeUa3aT/k5cqmsMv6BLR83L0JWTKJd4Z4NTtyjQnfTCgSjTTM= 
[root@head ~]# ssh c1
The authenticity of host 'c1 (10.5.1.1)' can't be established.
ED25519 key fingerprint is SHA256:PhJ/66DIOePxyt/Bup2tHzg1wLntLb5IFlU+U4BLOUU.
This key is not known by any other names
Are you sure you want to continue connecting (yes/no/[fingerprint])? yes
Warning: Permanently added 'c1' (ED25519) to the list of known hosts.
[root@c1 ~]# 
```

```
% ssh rocky@[REDACTED]
Last login: Wed Oct 30 12:34:34 2024 from 2001:468:f01:1505:79ec:464f:b9bb:84a9
Configuring SSH for cluster access
[rocky@head ~]$ cat .ssh/authorized_keys 
[REDACTED]
ssh-rsa AAAAB3NzaC1yc2EAAAADAQABAAABgQDKzggfYEzXjTrijCRdzrG/nplavxv5jGTH4x12XbD+xHig8UghZuw13zl/jwY0uZib0yazh8O5kk8/w3b+J8zOHswGBS+x0V7GZQvcoX9zo/9B5af40nyM218t/rs3RE5RQM+yoonyhJCH1kUMGJIa7zcERXkaPOTaGmTzFluKqK9pZ2RhC5phoN3TYnfVt/jCd9aFrWfqFtmxOEVaDL/XNB14PEadYJ8ysXXCjnlq5HVroUN30XUOaQD3k/9Pfb9kP06M1baeVdzpSdobYjGdLSDBILD/KKBnlfDtJs7x8/Tp0fyqmgUc8VYg/EygV9cdZ6cUsvOxo/yStFGcHiMlvB/l7OfwGhaZE2Oqb4v/Pm7sU5q2Zp1xP4aiSXmPoMjf55Z+0IrXhe4QILihvN5EemKhp/7fTniGiQHdjJIPCPRTklruzRuaPnj9CR2w8mjPZADlI1P2mERsLkCXw8JxZFqNbZ9FP5geChpoEBpLZh69mzKyY8IpyGRG9u9gl8E= Warewulf Cluster key
[rocky@head ~]$ ssh c1
Warning: Permanently added 'c1' (ED25519) to the list of known hosts.
Access denied: user rocky (uid=1000) has no active jobs on this node.
Connection closed by 10.5.1.1 port 22
[rocky@head ~]$ ssh-add -L
Could not open a connection to your authentication agent.
[rocky@head ~]$ 
```

Reply:
Interesting, now it fails after I `sudo -i`, wonder if it is a timeout or some overlay getting applied.

Reply:
Is there something in your nodes PAM configuration which returns the "Access denied"?

Reply:
Found my issue with root access.  When generating the Warewulf keys it updates /root/.ssh/authorized keys, which is included by the overlay, so a `wwctl overlay build` is needed after that.  

I have not touched pam.  I just added the slurm_pam back in.

Reply:
Removing slurm_pam allows me to login to a node again as a user.